setup/ci/codebuild.md

```diff
@@ -0,0 +1,121 @@
+
+hal config ci codebuild enable
+
+hal deploy apply
+```
+
+## Configure an AWS CodeBuild stage
+
+To run an AWS CodeBuild build as part of a Spinnaker pipeline:
+
+1. Create a stage of type *AWS CodeBuild*.
+
+2. Configure the stage by selecting the AWS CodeBuild account to use to run the build, and selecting the project name
+from the dropdown list.
+
+3. (Optional) In the *Source Configuration* section, you might 
+  - Select the source artifact to use as build source. If not specified, the source configured in project is used.
+  - Specify the [source version](https://docs.aws.amazon.com/codebuild/latest/APIReference/API_StartBuild.html#CodeBuild-StartBuild-request-sourceVersion)
+  of the build source.If not specified, the latest version is used.
+  - Specify the buildspec file. If left blank, the buildspec configured in project is used.
+  - Select secondary build sources. If not specified, the secondary sources configured in project is used
+
+4. (Optional) In the *Environment Configuration* section, you might specify the image tag or image digest that identifies the Docker image
+to use. If not specified, the image configured in project is used. **Note:** As prerequisite, follow this
+[user guide](https://docs.aws.amazon.com/codebuild/latest/userguide/sample-private-registry.html) to set up your registry credentials
+in project configuration.
+
+5. (Optional) In the *Advanced Configuration* section, you might specify environment variables to use in the build.
+Only environment variables in plain text are supported.
+
+6. (Optional) In the *Produces Artifacts* section, you might supply any artifacts that you expect the build to create in order to
+make these artifacts available to downstream stages. AWS CodeBuild supports creating S3 artifacts, which will be converted
+to Spinnaker artifacts and injected into the pipeline on completion of the build.
+
+While your build is executing, the stage details provide the following information
+* Current status of the build
+* ARN of the build
+* The link to view the build in the AWS CodeBuild Console
+* The link to view the build logs in AWS CloudWatch Logs
+* The link to view the build logs in S3
+
+## Configure your pipeline trigger (Optional)
+
+Configure your pipeline to be triggered by AWS CodeBuild build phase or state changes:
+
+1. Follow [instructions](https://docs.aws.amazon.com/codestar-notifications/latest/userguide/getting-started-build.html) to
+set up a notification rule for your project.
+
+2. Follow [instructions](/setup/triggers/amazon/) to create an Amazon Pub/Sub trigger.
+  - Skip the step to create SNS topic, as we will be using the SNS topic created in step 1.
+  - Create an SQS queue and subscribe to the SNS topic.
+  - Skip the step to create S3 bucket, as the notification will be sent from CodeBuild instead of S3.
+  - Write Echo configuration and set up the Amazon Pub/Sub trigger in UI
+
+3. Start a build in CodeBuild and your pipeline should be triggered.
```

Comment:
Pipeline is triggered only when the build completes?

Reply:
Pipeline could be triggered by either build phase or build state changes

Reply:
or both

Reply:
That's dependent on how they setup their notification rule, right?  It seems like it would only make sense if the pipeline triggered when the build completes so you can get the built artifacts later in the pipeline.  A suggestion of how to configure the notification rule to achieve that behavior would be useful here.

Reply:
Yes it depends on how they setup the notification rule. I will add steps here.